docs/JAX_for_PyTorch_users.md

```diff
@@ -0,0 +1,890 @@
+  for step, batch in enumerate(train_loader):
+
+    loss = train_step(model, optimizer, metrics, batch)
+
+    print(f"\r[train] epoch: {epoch + 1}/{num_epochs}, iteration: {step}, batch loss: {loss.item():.4f}", end="")
+  print("\r", end="")
+
+  for metric, value in metrics.compute().items():  # Compute the metrics.
+    metrics_history[f'train_{metric}'].append(value)  # Record the metrics.
+  metrics.reset()  # Reset the metrics for the test set.
+
+  # Compute the metrics on the test set after each training epoch.
+  model.eval()  # Set model to evaluation model: e.g. use stored batch statistics
+  for test_batch in test_loader:
+    eval_step(model, metrics, test_batch)
+
+  # Log the test metrics.
+  for metric, value in metrics.compute().items():
+    metrics_history[f'test_{metric}'].append(value)
+  metrics.reset()  # Reset the metrics for the next training epoch.
+
+  print(
+    f"[train] epoch: {epoch + 1}/{num_epochs}, "
+    f"loss: {metrics_history['train_loss'][-1]:0.4f}, "
+    f" accuracy: {metrics_history['train_accuracy'][-1] * 100:0.4f}"
+  )
+  print(
+    f"[test] epoch: {epoch + 1}/{num_epochs}, "
+    f"loss: {metrics_history['test_loss'][-1]:0.4f}, "
+    f"accuracy: {metrics_history['test_accuracy'][-1] * 100:0.4f}"
+    "\n"
+  )
+```
+
+### Further reading
+
+More details about Flax NNX API, how to save and load the model's state and about available optimizers, we suggest to check out the links below:
+- [FLAX NNX](https://flax.readthedocs.io/en/latest/nnx_basics.html)
+- [Save & Load model's state](https://flax.readthedocs.io/en/latest/guides/checkpointing.html)
+- [Optax](https://optax.readthedocs.io/en/latest/)
+
+
+Other AI/ML tutorials to check out:
+- [JAX AI Stack tutorials](https://jax-ai-stack.readthedocs.io/en/latest/tutorials.html)
+
+```{code-cell} ipython3
+
+```
+
+```{code-cell} ipython3
+
+```
+
+## Just-In-Time (JIT) compilation in JAX
+
+
+PyTorch users know very well about the eager mode execution of the operations in PyTorch, e.g. the operations are executed one by one without any high-level optimizations on sets of operations. Similarly, almost everywhere in this tutorial we used JAX in the eager mode as well.
+
+
+In PyTorch 1.0 [TorchScript](https://pytorch.org/docs/stable/jit.html) was introduced to optimize and serialize PyTorch models by capturing the execution graph into TorchScript programs, which can then be run independently from Python, e.g. as a C++ program.
+
+
+In JAX, there is a similar transformation: [`jax.jit`](https://jax.readthedocs.io/en/latest/_autosummary/jax.jit.html#jax.jit). It performs JIT compilation of a Python function for efficient execution in XLA. Behind the scenes, `jax.jit` is tracing the function wrapping the inputs into tracers to record all JAX operations. By default, JAX JIT is compiling the function on the first call and reusing the cached compiled XLA code on subsequent calls.
```

Comment:
"tracing the function wrappoing the inputs into tracers" sounds strange here. Is there a missing "and"?